### PR TITLE
Adds support for includeConceptDescription parameter for Serialization interface

### PR DESCRIPTION
**Pull request by mdanish98**
Signed-off-by: Mohammad Ghazanfar Ali Danish <ghazanfar.danish@iese.fraunhofer.de>

### CODE REVIEW

File: basyx.aasenvironment/basyx.aasenvironment-core/src/main/java/org/eclipse/digitaltwin/basyx/aasenvironment/AasEnvironmentSerialization.java [hunk 1]
```diff
@@ -42,12 +42,13 @@ public interface AasEnvironmentSerialization {
 	 * 
 	 * @param aasIds
 	 * @param submodelIds
+	 * @param includeConceptDescriptions
 	 * @return
 	 * @throws SerializationException
 	 */
-	public String createJSONAASEnvironmentSerialization(List<String> aasIds, List<String> submodelIds) throws SerializationException;
+	public String createJSONAASEnvironmentSerialization(List<String> aasIds, List<String> submodelIds, Boolean includeConceptDescriptions) throws SerializationException;
```
Comment (FrankSchnicke) [inline basyx.aasenvironment/basyx.aasenvironment-core/src/main/java/org/eclipse/digitaltwin/basyx/aasenvironment/AasEnvironmentSerialization.java:49]: Why are you using Boolean instead of boolean here?

File: basyx.aasenvironment/basyx.aasenvironment-http/src/test/java/org/eclipse/basyx/digitaltwin/aasenvironment/http/TestAasEnvironmentHTTP.java [hunk 2]
```diff
@@ -36,61 +36,83 @@ public class TestAasEnvironmentHTTP {
 	private static final String ACCEPT_XML = "application/xml";
 	private static final String ACCEPT_AASX = "application/asset-administration-shell-package+xml";
 
-	private static ConfigurableApplicationContext appContext;
+	private ConfigurableApplicationContext appContext;
 
-	@BeforeClass
-	public static void startAasRepo() throws Exception {
+	@Before
```
Comment (FrankSchnicke) [inline basyx.aasenvironment/basyx.aasenvironment-http/src/test/java/org/eclipse/basyx/digitaltwin/aasenvironment/http/TestAasEnvironmentHTTP.java:41]: Why did you change this to @Before?

File: basyx.aasenvironment/basyx.aasenvironment-http/src/test/java/org/eclipse/basyx/digitaltwin/aasenvironment/http/TestAasEnvironmentHTTP.java [hunk 3]
```diff
@@ -140,8 +162,8 @@ public String readStringFromFile(String fileName) throws FileNotFoundException,
 		return IOUtils.toString(in, StandardCharsets.UTF_8.name());
 	}
 
-	@AfterClass
-	public static void shutdown() {
+	@After
```
Comment (FrankSchnicke) [inline basyx.aasenvironment/basyx.aasenvironment-http/src/test/java/org/eclipse/basyx/digitaltwin/aasenvironment/http/TestAasEnvironmentHTTP.java:165]: Why did you change this to @After?

File: basyx.aasenvironment/basyx.aasenvironment-core/src/main/java/org/eclipse/digitaltwin/basyx/aasenvironment/ConceptDescriptionIdCollector.java
Comment (FrankSchnicke) [inline basyx.aasenvironment/basyx.aasenvironment-core/src/main/java/org/eclipse/digitaltwin/basyx/aasenvironment/ConceptDescriptionIdCollector.java:1]: I'm missing tests that showcase that this class is working correctly. Can you add them?

**Review by FrankSchnicke — CHANGES_REQUESTED** [review]

File: basyx.aasenvironment/basyx.aasenvironment-core/src/main/java/org/eclipse/digitaltwin/basyx/aasenvironment/AasEnvironmentSerialization.java (hunk 1 above)
Comment (mdanish98) [reply basyx.aasenvironment/basyx.aasenvironment-core/src/main/java/org/eclipse/digitaltwin/basyx/aasenvironment/AasEnvironmentSerialization.java]: Replaced the Boolean wrapper class with the primitive

File: basyx.aasenvironment/basyx.aasenvironment-http/src/test/java/org/eclipse/basyx/digitaltwin/aasenvironment/http/TestAasEnvironmentHTTP.java (hunk 2 above)
Comment (mdanish98) [reply basyx.aasenvironment/basyx.aasenvironment-http/src/test/java/org/eclipse/basyx/digitaltwin/aasenvironment/http/TestAasEnvironmentHTTP.java]: Revert this to the previous state i.e. @BeforeClass

File: basyx.aasenvironment/basyx.aasenvironment-http/src/test/java/org/eclipse/basyx/digitaltwin/aasenvironment/http/TestAasEnvironmentHTTP.java (hunk 3 above)
Comment (mdanish98) [reply basyx.aasenvironment/basyx.aasenvironment-http/src/test/java/org/eclipse/basyx/digitaltwin/aasenvironment/http/TestAasEnvironmentHTTP.java]: Reverted this to the previous state i.e. @AfterClass

File: basyx.aasenvironment/basyx.aasenvironment-core/src/main/java/org/eclipse/digitaltwin/basyx/aasenvironment/ConceptDescriptionIdCollector.java
Comment (mdanish98) [reply basyx.aasenvironment/basyx.aasenvironment-core/src/main/java/org/eclipse/digitaltwin/basyx/aasenvironment/ConceptDescriptionIdCollector.java:1]: Added a test class to showcase the correct working of ConceptDescriptionIdCollector

File: pom.xml [hunk 4]
```diff
@@ -250,6 +250,17 @@
 					</exclusion>
 				</exclusions>
 			</dependency>
+			<dependency>
+				<groupId>org.eclipse.digitaltwin.aas4j</groupId>
+				<artifactId>dataformat-core</artifactId>
+				<version>1.0.0-milestone-03</version>
```
Comment (FrankSchnicke) [inline pom.xml:256]: Please use ${aas4j-version}

File: basyx.aasenvironment/basyx.aasenvironment-core/src/main/java/org/eclipse/digitaltwin/basyx/aasenvironment/base/DefaultAASEnvironmentSerialization.java [hunk 5]
```diff
@@ -37,54 +39,101 @@
...
+	public byte[] createAASXAASEnvironmentSerialization(@Valid List<String> aasIds, @Valid List<String> submodelIds, @Valid boolean includeConceptDescriptions) throws SerializationException, IOException {
+		Environment aasEnvironment = extractShellsAndSubmodels(aasIds, submodelIds, includeConceptDescriptions);
 
 		ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
 		aasxSerializer.write(aasEnvironment, null, outputStream);
 		return outputStream.toByteArray();
 	}
 
-	private Environment extractShellsAndSubmodels(List<String> aasIds, List<String> submodelIds) {
+	private Environment extractShellsAndSubmodels(List<String> aasIds, List<String> submodelIds, boolean includeConceptDescriptions) {
```
Comment (FrankSchnicke) [inline basyx.aasenvironment/basyx.aasenvironment-core/src/main/java/org/eclipse/digitaltwin/basyx/aasenvironment/base/DefaultAASEnvironmentSerialization.java:102]: The function name does reflect the intent anymore since now conceptDescriptions may be included

File: basyx.aasenvironment/basyx.aasenvironment-core/src/test/java/org/eclipse/digitaltwin/basyx/aasenvironment/TestConceptDescriptionIdCollector.java [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+import org.eclipse.digitaltwin.aas4j.v3.model.SubmodelElementCollection;
+import org.eclipse.digitaltwin.aas4j.v3.model.impl.DefaultAssetAdministrationShell;
+import org.eclipse.digitaltwin.aas4j.v3.model.impl.DefaultEnvironment;
+import org.eclipse.digitaltwin.aas4j.v3.model.impl.DefaultKey;
+import org.eclipse.digitaltwin.aas4j.v3.model.impl.DefaultProperty;
+import org.eclipse.digitaltwin.aas4j.v3.model.impl.DefaultReference;
+import org.eclipse.digitaltwin.aas4j.v3.model.impl.DefaultSubmodel;
+import org.eclipse.digitaltwin.aas4j.v3.model.impl.DefaultSubmodelElementCollection;
+import org.eclipse.digitaltwin.basyx.submodelservice.DummySubmodelFactory;
+import org.junit.jupiter.api.BeforeEach;
```
Comment (FrankSchnicke) [inline basyx.aasenvironment/basyx.aasenvironment-core/src/test/java/org/eclipse/digitaltwin/basyx/aasenvironment/TestConceptDescriptionIdCollector.java:23]: We're using JUnit 4 for now. This is JUnit 5

**Review by FrankSchnicke — CHANGES_REQUESTED** [review]

**Review by FrankSchnicke — APPROVED** [review]
